yellow-paper/docs/cryptography/performance-targets.md

```diff
@@ -0,0 +1,121 @@
+
+ "MVP" = minimum standards that we can go to main-net with.
+
+| metric | how to measure | MVP | ideal |
+| --- | --- | --- | --- |
+| proof size | total size of a user tx incl. goblin plonk proofs | 32kb | 8kb |
+| prover time | 8 iterations of protogalaxy with 2^17 circuits (web browser) | 1 min | 10 seconds |
+| verifier time | how long does it take the verifier to check a proof (incl. grumpkin IPA MSMs) | 25ms | 1ms |
+| memory consumption | fold 2^17 circuits into an accumulator an arbitrary number of times | 2gb | 512mb |
+| size of the kernel circuit | number of gates | 2^17 | 2^15 |
+| Aztec Virtual Machine prover time | 1 million VM step circuit | 30 seconds | 5 seconds |
+| Aztec Virtual Machine memory consumption | 1 million VM step circuit | 128Gb | 16Gb |
+
+### Proof size
+
+At a tx throughput of 1,024 tx per second, each Aztec node (not sequencer/prover, just a regular node that is sending transactions) needs to download `1024*proof_size` bytes of data to keep track of the mempool. 32kb proofs = 32MB per second which is on the threshold of practical outside of a datacenter. 8MB per second is much more practical.
+
+At launch, the throughput won't be this high, but ideally we architect Aztec such that the network can scale effectively by throwing hardware+resources at bottlenecks (easy to do in a post-decentralisation), without protocol, architecture or tech upgrades ( harder post-decentralisation).
+
+### Prover time
+
+The critical UX factor. 1 minute per user tx is on the cusp of workable. The faster we can get this the better our network's value proposition for users and developers
+
+### Verifier time
+
+This matters because verifying a transaction is effectively free work being performed by sequencers and network nodes that propagate txns to the mempool. If verification time becomes too large it opens up potential DDOS attacks.
+
+At a 1,024 transactions per second and 20ms verification time, a block builder needs to run 20 machines in parallel just to verify natively all of the proofs in a block! Not ideal but probably tolerable given this problem has a "more hardware go brrrr" solution.
+
+### Memory consumption
+
+This is *critical*. Users can tolerate slow proofs, but if Honk consumes too much memory, a user cannot make a proof at all.
+
+safari on iPhone will purge tabs that consume more than 1gb of RAM. 2gb will work in web browsers and some phones (but not all). 512mb will allow most phones to make proofs.
+
+### Kernel circuit size
+
+Not a critical metric, but the prover time + prover memory metrics are predicated on a kernel circuit costing about 2^17 constraints!
+
+### AVM Prover time
+
+Our goal is to hit main-net with a network that can support 10 transactions per second. We need to estimate how many VM computation steps will be needed per transaction to determine the required speed of the VM Prover. The following uses very conservative estimations due to the difficulty of estimating this.
```

Comment:
> 10 transactions per second

How is "transaction" defined here? How many new notes/nullifiers? How many enqueued public function calls?

Reply:
Good question. I would prefer we write a baseline aztec-connect style defi interaction contract and use that as the baseline for a "transaction". It will produce more conservative estimates vs a basic transfer, but I suspect we will over-fit to benchmarks if we measure against a transaction type that does not make many function calls.